[Doc] Add "Setting the version tracking metadata for cops" section (#13660)
rubocop/rubocop: docs/modules/ROOT/pages/configuration.adoc

  DocumentationExtension: .md
----

== Setting the version tracking metadata for cops

> Each cop should gain the following additional metadata:

Reviewer: ```suggestion
Each cop can have the following additional metadata:
```

It might be useful to note that `VersionAdded` is required but the others are contingent.

* `VersionAdded` - the RuboCop version in which it was added
* `VersionChanged` - the RuboCop version in which it was changed in a user-impacting way (new config, updated defaults, etc)

> * `VersionRemoved` - the RuboCop version in which the cop was removed for whatever reason.

Reviewer: We don't seem to have used `VersionRemoved` anywhere, and it is not referenced anywhere other than being a valid configuration parameter. Perhaps it's useful to explain why it would be set? Why would we retain config for a cop that is removed?

Author: `VersionRemoved` was removed because it cannot be configured in `default.yml` and I added it along with a setting example for illustration.

> Those will be pretty useful for the documentation (so the manual generation has to be enhanced to include them) and keeping track of changes.

Reviewer: This line probably doesn't make sense in user-facing documentation.

Author: Yeah, on the other hand, this might be a meaningful configuration for users who distribute custom cops as a gem. I considered adding it to development.adoc, but thought it might be more appropriate to include it in configuration.adoc for this reason.

At the beginning of the section, I added: "This configuration is particularly useful when custom cops are distributed as a gem."

Reviewer 2: Yeah, we can keep this. The `VersionAdded` would also be useful for the long-demanded and stuck option to enable cops introduced up to a certain version (https://github.com/rubocop/rubocop/issues/11875).
